micrometer-core/src/main/java/io/micrometer/core/instrument/binder/cache/CaffeineCacheMetrics.java

```diff
@@ -67,7 +67,7 @@ public CaffeineCacheMetrics(Cache<?, ?> cache, String cacheName, Iterable<Tag> t
      * @param <C>       The cache type.
      * @return The instrumented cache, unchanged. The original cache is not wrapped or proxied in any way.
      */
-    public static <C extends Cache> C monitor(MeterRegistry registry, C cache, String cacheName, String... tags) {
+    public static <C extends Cache<?, ?>> C monitor(MeterRegistry registry, C cache, String cacheName, String... tags) {
```

Comment:
Could you help me understand in what circumstances this helps? It doesn't help the compiler infer generic type arguments on an expression involving `monitor` is evaluated, and the lack of wildcards doesn't generate compiler warnings about missing type arguments.

Reply:
With out parametrization it actually generates compiler warnings
![image](https://user-images.githubusercontent.com/4628962/50566126-db249880-0cea-11e9-8259-27d0d6cb66e9.png)

Apart from addressing warnings, it also makes implementation consistent across different cache metrics. Here are few examples where it's already parametrized:
https://github.com/micrometer-metrics/micrometer/blob/master/micrometer-core/src/main/java/io/micrometer/core/instrument/binder/cache/CaffeineCacheMetrics.java#L54
https://github.com/micrometer-metrics/micrometer/blob/master/micrometer-core/src/main/java/io/micrometer/core/instrument/binder/cache/HazelcastCacheMetrics.java#L62
https://github.com/micrometer-metrics/micrometer/blob/master/micrometer-core/src/main/java/io/micrometer/core/instrument/binder/cache/HazelcastCacheMetrics.java#L47
https://github.com/micrometer-metrics/micrometer/blob/master/micrometer-core/src/main/java/io/micrometer/core/instrument/binder/cache/JCacheMetrics.java#L58
https://github.com/micrometer-metrics/micrometer/blob/master/micrometer-core/src/main/java/io/micrometer/core/instrument/binder/cache/JCacheMetrics.java#L73

Reply:
Thanks. Ah Eclipse. I think it just isn't smart enough to tell that in this context the generic type parameters don't actually help the compiler. IntelliJ doesn't show the same warnings. I'm happy to leave the type wildcards to help our Eclipse friends.